lib/OpenQA/Shared/Controller/Session.pm

```diff
@@ -21,13 +21,14 @@ use Carp 'croak';
 sub ensure_user {
     my ($self) = @_;
     return 1 if $self->current_user;
-    $self->redirect_to('login');
+    $self->redirect_to($self->url_for('login')->query(return_page => $self->req->url));
```

Comment:
Because only the "page" within the current openQA instance is relevant here I'd strip the other URL parts here right away:
```suggestion
    $self->redirect_to($self->url_for('login')->query(return_page => $self->req->url->path_query));
```

The same counts for the other occurrences. It would actually make sense to extract this into another function to have only one occurrence of this code.

Reply:
That will already happen in `lib/OpenQA/WebAPI/Auth/OpenID.pm` so I thought it shouldn't be done twice.
Also when doing some tests I found out that it is not needed - it already redirects you using

```
Location: /login?return_page=%2Fadmin%2Fusers
```